### PR TITLE
Fixes find widget shall be shown per split terminal

### LINKED ISSUE #145865: Find widget shall be shown per split terminal
Testing #145629

image

Find widget is shown for all split terminals which makes it confusing where I am searching. For eg., in above screenshot the wrong terminal is highlighting find results which is not matched with with text in find widget.

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #145865

### CODE REVIEW

**Comment by meganrogge** [comment]
Thanks @jeanp413 for working on this. The code changes/clean-up look great 👍🏼 

Github won't let me attach images or gifs ATM 😢 

I notice a few things:

1. You have to focus a terminal for that find widget to become active. Maybe this is what we want/ as designed or maybe the better approach would be to set the terminal as the active instance when its find widget is focused - not sure. It does feel cumbersome to have to focus the terminal first though IMO.

2. When minding #1, focusing terminal A and pressing enter in the find widget a few times results in the index `1 of 4` for example updating correctly ✅ . Leaving that find widget open and focusing terminal B and doing the same the results in the index being out of sync with the currently selected element.

**Comment by jeanp413** [comment]
@meganrogge updated the PR, last commit if optional just some :lipstick: to match the findwidget behavior from a normal editor
Current behavior in stable:
terminal_find_widget

**Comment by meganrogge** [comment]
great @jeanp413 thanks. I was just working on fixing the two problems I mentioned above and will revert my fix as yours looks better 👍🏼

File: src/vs/workbench/contrib/terminal/common/terminalContextKey.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export namespace TerminalContextKeys {
 	/** Whether the active terminal's find widget text input is focused. */
 	export const findInputFocus = new RawContextKey<boolean>(TerminalContextKeyStrings.FindInputFocused, false, true);
 
-	/** Whether an element iwhtin the active terminal's find widget is focused. */
+	/** Whether an element whitin the active terminal's find widget is focused. */
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/common/terminalContextKey.ts:110]:
```suggestion
	/** Whether an element within the active terminal's find widget is focused. */
```

**Review by meganrogge — DISMISSED** [review]
Works great 😄

**Review by Tyriar — APPROVED** [review]
Works great, sorry about the delay 🙂

**Pull request merged**